primo/data_parser/well_data.py

```diff
@@ -0,0 +1,1149 @@
+                # This will not happen for supported metrics, because this check
+                # is performed before in col_names.check_columns_available()
+                # method. However, this may happen for non-supported/user-defined
+                # custom metrics.
+                msg = (
+                    f"data_col_name attribute for metric {metric.name}/{metric.full_name} "
+                    f"is not provided."
+                )
+                raise_exception(msg, ValueError)
+
+            # Step 2: Fill incomplete data
+            self.fill_incomplete_data(
+                col_name=metric.data_col_name,
+                value=metric.fill_missing_value,
+                flag_col_name=metric.name + "_flag",
+            )
+
+            # Step 3: If it is a binary-type metric, then convert the data to 0-1
+            if metric.is_binary_type:
+                self.convert_data_to_binary(metric.data_col_name)
+
+            # Step 4: Ensure that the column in numeric
+            flag, non_numeric_rows = self.is_data_numeric(col_name=metric.data_col_name)
+            if not flag:
+                msg = (
+                    f"Unable to compute scores for metric {metric.name}/"
+                    f"{metric.full_name},  because the column {metric.data_col_name} "
+                    f"contains non-numeric values in rows {non_numeric_rows}."
+                )
+                raise_exception(msg, ValueError)
+
+            # Step 5: Compute the priority score
+            max_value = self.data[metric.data_col_name].max()
+            min_value = self.data[metric.data_col_name].min()
+
+            # Check if division by a zero is likely
+            if np.isclose(max_value, min_value, rtol=0.001):
+                # All cells in this column have equal value.
+                # To avoid division by zero, set min_value = 0
+                min_value = 0
+
+            if np.isclose(max_value, 0, rtol=0.001):
+                # All values in this column are likely zeros.
+                # To avoid division by zero, set max_value = 1
+                max_value = 1.0
+
+            if metric.has_inverse_priority:
+                self.data[metric.score_col_name] = (
+                    (max_value - self.data[metric.data_col_name])
+                    / (max_value - min_value)
+                ) * metric.effective_weight
+
+            else:
+                self.data[metric.score_col_name] = (
+                    (self.data[metric.data_col_name] - min_value)
+                    / (max_value - min_value)
+                ) * metric.effective_weight
+
+        LOGGER.info("Computing the total priority score.")
+        self.data["Priority Score [0-100]"] = self.data[
+            self.get_priority_score_columns
+        ].sum(axis=1)
+
+        self.check_data_in_range("Priority Score [0-100]", 0.0, 100.0)
+        LOGGER.info("Completed the calculation of priority scores.")
+
+    @property
```

Comment:
I would move these attributes up to the top of the file with the __init__ and the other attributes. It makes finding the attribute definitions easier for someone using the class. But this is completely my preference.

Reply:
Made the change. Moved the properties to the top.